### ISSUE TITLE
Can't quite override modules database

### ISSUE THREAD

**Issue by paxtonhare** [rjrudin/ml-gradle#78]
Let's say I want to override the name of my modules db:

```groovy
ext {
    // mlAppConfig is an instance of com.marklogic.appdeployer.AppConfig
    mlAppConfig {
        modulesDatabaseName = "my-modules-db-name"
    }
}
````

When I run ./gradlew mlLoadModules I see that it's trying to use the default modules db instead of mine:

```
11:44:38.875 [DEBUG] [com.marklogic.client.modulesloader.impl.XccAssetLoader] Initializing XCC session; host: 10.211.55.11; username: admin; database name: advanced-gradle-example-modules
```

I think I have tracked the issue down to line 337 in this line in MarkLogicPlugin.groovy.
```groovy
        // Modules
        LoadModulesCommand lmc = new LoadModulesCommand()

        lmc.initializeDefaultModulesLoader(context) // this line!

        project.extensions.add("mlLoadModulesCommand", lmc)
        commands.add(lmc)
```

I'm thinking that line 337 is getting run before i get a chance to alter the mlAppConfig variable in my gradle.properties file.

So I did this and it serves as a temporary workaround:

```groovy

ext {
    // mlAppConfig is an instance of com.marklogic.appdeployer.AppConfig
    mlAppConfig {
        modulesDatabaseName = "my-modules-db-name"
    }
}

ext {
    def lmc = mlAppDeployer.getCommand("LoadModulesCommand")
    lmc.setModulesLoader(new com.marklogic.client.modulesloader.impl.DefaultModulesLoader(mlAppConfig.newXccAssetLoader()))
}
```

**Comment by rjrudin** [comment]
That workaround will do the trick (nice one!). I think the solution is for me to introduce a property for the modules database name, as the properties are processed before the plugin starts creating commands. 

So most likely:

mlModulesDatabaseName=my-name

I put this into 2.1.1. It'll need a fix in ml-app-deployer, which has the code for processing properties from any source.

**Comment by rjrudin** [comment]
I forgot about this ticket and created https://github.com/rjrudin/ml-gradle/issues/91 instead - this is now fixed in ml-gradle, and ml-app-deployer 2.2.0 will also have an "mlModulesDatabaseName" property so you can set it in gradle.properties.
